web/src/main/java/se/inera/intyg/intygsadmin/web/controller/PrivatePractitionerController.java

```diff
@@ -18,66 +18,43 @@
  */
 package se.inera.intyg.intygsadmin.web.controller;
 
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpHeaders;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
+import se.inera.intyg.intygsadmin.web.controller.dto.PrivatePractitionerDTO;
 import se.inera.intyg.intygsadmin.web.integration.PPIntegrationRestService;
-import se.inera.intyg.intygsadmin.web.integration.model.PrivatePractitioner;
-import se.inera.intyg.intygsadmin.web.util.PrivatePractitionerFileWriter;
+import se.inera.intyg.intygsadmin.web.service.PrivatePractitionerService;
 
 @RestController
 @RequestMapping("/api/privatepractitioner")
 public class PrivatePractitionerController {
 
-    private PPIntegrationRestService ppIntegratedUnitsService;
+    private final PPIntegrationRestService ppIntegrationRestService;
+    private final PrivatePractitionerService privatePractitionerService;
 
-    @Autowired
-    public PrivatePractitionerController(PPIntegrationRestService ppIntegratedUnitsService) {
-        this.ppIntegratedUnitsService = ppIntegratedUnitsService;
+    public PrivatePractitionerController(PPIntegrationRestService ppIntegrationRestService,
+        PrivatePractitionerService privatePractitionerService) {
+        this.ppIntegrationRestService = ppIntegrationRestService;
+        this.privatePractitionerService = privatePractitionerService;
     }
 
     @GetMapping("/{personOrHsaId}")
-    public ResponseEntity<PrivatePractitioner> getPrivatePractitioner(@PathVariable String personOrHsaId) {
-
-        PrivatePractitioner privatePractitioner = ppIntegratedUnitsService.getPrivatePractitioner(personOrHsaId);
-        if (privatePractitioner == null) {
-            return ResponseEntity.notFound().build();
-        }
-
-        return ResponseEntity.ok(privatePractitioner);
+    public ResponseEntity<PrivatePractitionerDTO> getPrivatePractitioner(@PathVariable String personOrHsaId) {
+        return privatePractitionerService.getPrivatePractitioner(personOrHsaId);
     }
 
     @GetMapping("/file")
     public ResponseEntity<byte[]> getPrivatePractitionerFile() {
+        return privatePractitionerService.getPrivatePractitionerFile();
+    }
 
-        List<PrivatePractitioner> privatePractitionerList = ppIntegratedUnitsService.getAllPrivatePractitioners();
```

Comment:
Why was this removed?

Reply:
I didn't remove it. But since I thought that the controller had too much logic for a controller I moved this to the new service that I created, together with the updated getPrivatePractitioner-function. The behaviour has not changed.